stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/ChatSocketServiceImpl.kt

```diff
@@ -170,28 +174,36 @@ internal class ChatSocketServiceImpl private constructor(
     }
 
     private fun reconnect(connectionConf: ConnectionConf) {
-        releaseSocket()
+        shutdownSocketConnection()
         setupSocket(connectionConf)
     }
 
     private fun setupSocket(connectionConf: ConnectionConf) {
         logger.logI("setupSocket")
-        when (connectionConf) {
-            is ConnectionConf.None -> {
-                state = State.DisconnectedPermanently
-            }
-            is ConnectionConf.AnonymousConnectionConf -> {
-                state = State.Connecting
-                socket = socketFactory.createAnonymousSocket(connectionConf.endpoint, connectionConf.apiKey)
-            }
-            is ConnectionConf.UserConnectionConf -> {
-                state = State.Connecting
-                socket = socketFactory.createNormalSocket(connectionConf.endpoint, connectionConf.apiKey, connectionConf.user)
-            }
-        }.exhaustive
+        with(connectionConf) {
+            when (this) {
+                is ConnectionConf.None -> {
+                    state = State.DisconnectedPermanently
+                }
+                is ConnectionConf.AnonymousConnectionConf -> {
+                    state = State.Connecting
+                    socket = socketFactory.createAnonymousSocket(endpoint, apiKey)
+                }
+                is ConnectionConf.UserConnectionConf -> {
+                    state = State.Connecting
+                    socketConnectionJob = GlobalScope.launch(DispatcherProvider.IO) {
+                        tokenManager.ensureTokenLoaded()
+                        withContext(DispatcherProvider.Main) {
+                            socket = socketFactory.createNormalSocket(endpoint, apiKey, user)
+                        }
+                    }
+                }
```

Comment:
The job needs to cancelled as `tokenManager` can load token for quite a while and the following will have no effect
```
 private fun shutdownSocketConnection() {
        socket?.close(1000, "Connection close by client")	       
        socket = null	       
    }	
```